### ISSUE TITLE
Support server.compression for reactive servers

### ISSUE THREAD

**Issue by unlimitedsola** [spring-projects/spring-boot#10782]
According to https://docs.spring.io/spring-boot/docs/current-SNAPSHOT/reference/htmlsingle/#how-to-enable-http-response-compression
webflux with netty server currently doesn't support compression.

**Comment by snicoll** [comment]
@bclozel given that it's blocked and not scheduled for `reactor-netty` 0.8.0, I think this one should be postponed.

**Comment by bclozel** [comment]
@snicoll reactor-netty supports server compression with only one option: the minimum response size (see [HttpServerOptions](https://github.com/reactor/reactor-netty/blob/master/src/main/java/reactor/ipc/netty/http/server/HttpServerOptions.java)). We can provide that support.

To be on par with the other servers, we'd need reactor/reactor-netty#218.

**Comment by bclozel** [comment]
We decided to go ahead and support compression with just the minimum size option available.

**Comment by unlimitedsola** [comment]
It seems that `reactor-netty` team is not actively fixing issues which makes me a bit disappointed

**Comment by ealexhaywood** [comment]
Can you provide an example of a Reactive `@RestController` to get it to properly gzip responses?  Setting `server.compression.enabled=true` doesn't seem to do anything but add a `content-encoding: gzip` response header.  The response itself is not gzipped.  

I have an endpoint that returns a gzip file that I was compressing myself using `GZIPOutputStream` like so:

```java
@GetMapping(path = "/something/{variable}", produces = "application/gzip")
    public ResponseEntity<Flux<byte[]>> getGzipFile(@PathVariable("variable") String variable) {
        Flux<byte[]> results = generatorService.generateGzipFile(variable);
        return ResponseEntity.ok()
                .header("Content-Disposition", "attachment; filename=" + variable + ".gz")
                .body(results);
    }
```
This worked pretty well, but I figured I'd try to let the Netty handle the compression so I don't have to deal with managing streams, compression, and byte arrays myself.   

I took out the compression code in my `generatorService` and enabled the compression server setting and my responses were not gzipped.  I did add `application/gzip` as a mime type as well in the server compression settings.

**Comment by bclozel** [comment]
@ealexhaywood Could you create a new issue for that with a sample application showing this problem?

**Comment by wilkinsona** [comment]
I would guess that this behaviour is due to https://github.com/reactor/reactor-netty/issues/411.

**Comment by ealexhaywood** [comment]
Oh okay great, so it seems like it has been addressed and will be in your 0.7.9.RELEASE sometime soon.  About how long will it takes for this to make its way into a Spring Boot release?

**Comment by bclozel** [comment]
[Spring Boot 2.0.5 is scheduled for next week](https://github.com/spring-projects/spring-boot/milestone/115).
